mllib/src/main/scala/org/apache/spark/ml/feature/OneHotEncoder.scala

```diff
@@ -41,8 +41,12 @@ import org.apache.spark.sql.types.{DoubleType, NumericType, StructType}
  * The output vectors are sparse.
  *
  * @see `StringIndexer` for converting categorical values into category indices
+ * @deprecated `OneHotEncoderEstimator` will be renamed `OneHotEncoder` and this `OneHotEncoder`
```

Comment:
Note for the future: For 3.0, it'd be nice to do what you're describing here but also leave OneHotEncoderEstimator as a deprecated alias.  That way, user code won't break but will have deprecation warnings when upgrading to 3.0.

Reply:
Ok. Sounds good.